src/System.Text.Json/src/System/Text/Json/Serialization/ReflectionEmitMaterializer.cs

```diff
@@ -14,22 +14,41 @@ internal sealed class ReflectionEmitMaterializer : ClassMaterializer
         public override JsonClassInfo.ConstructorDelegate CreateConstructor(Type type)
         {
             Debug.Assert(type != null);
+            ConstructorInfo realMethod;
 
-            ConstructorInfo realMethod = type.GetConstructor(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance, binder: null, Type.EmptyTypes, modifiers: null);
-            if (realMethod == null)
+            if (type.IsValueType)
             {
-                return null;
+                realMethod = null;
+            }
+            else
+            {
+                realMethod = type.GetConstructor(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance, binder: null, Type.EmptyTypes, modifiers: null);
+                if (realMethod == null) return null;
             }
 
             var dynamicMethod = new DynamicMethod(
-                realMethod.Name,
-                type,
+                ".ctor",
+                typeof(object),
```

Comment:
Why use typeof(object) here?

Reply:
There is no need to introduce condition here. `JsonClassInfo.ConstructorDelegate` returns an `object`.